examples/example6.c++

```diff
@@ -0,0 +1,35 @@
+#define NONIUS_RUNNER
+#include <nonius.h++>
+
+#include <string>
+#include <vector>
+
+NONIUS_PARAM("size", 100u)
+NONIUS_PARAM("string", std::string{"foo"})
+
+NONIUS_BENCHMARK("push_back vector", [](nonius::chronometer meter)
+{
+    auto n = meter.param("size");
+    auto s = meter.param<std::string>("string");
```

Comment:
Lol, I just sent you an email on this topic. I copy here the proposal so other people can see and discuss what's best.

---

The current API forces the user to write the type the type and string name of the parameter on every benchmark.  That feels now too "dynamic", and was also preventing some more elegant implementations of type erasure could be otherwise added without this API.  Here is the change proposal:

**Option 1**:  macro that expands to a type:

```cpp
NONIUS_PARAM(N, std::size_t{10}) // defines a type N that encapsulates
                                 // the information about the parameter

NONIUS_BENCHMARK("foo", [] (nonius::parameters params) {
    auto n = params.get<N>();    // decltype(n) == std::size_t
})

```

**Option 2**: no macros, global object instead of type

```cpp
nonius::param<std::size_t> N{"N", 10};

NONIUS_BENCHMARK("foo", [] (nonius::parameters params) {
    auto n = params.get(N);      // decltype(n) == std::size_t
})
```

I was considering that the `N` object could itself provide access to the parameter value, but that would mean that global state is required and benchmarks could not be run in parallel (well, maybe yest with a `thread_local` global, still, global state makes me unconfortable :-).

Note that the two options can be combined.  Also, in either case, I-d like to add higher order functions to make param extraction terser. e.g.

```cpp
NONIUS_PARAM(N, std::size_t{10})

NONIUS_BENCHMARK("foo", nonius::parametrized<N>([] (auto n) {
    ...
})
```